### PR TITLE
fix: prevent ClassCastException when calling Tabs#add(Component...)

### LINKED ISSUE #4680: Tabs [Vaadin 24.0.0.beta1]: ClassCastException in public void add(Component... components)
### Description

`Tabs#add(Component... components)` takes a Component[] (as varargs) and tries to cast it to a Tab[] which is doomed to fail. The method is already deprecated (and I'll move away from it), but it should still probably either be fixed or removed.

### Expected outcome

No ClassCastException or remove the method.

### Minimal reproducible example

```java
public static void main(String[] args)
{
  Component[] tabsArray = {new Tab(), new Tab()};
  Tabs tabs = new Tabs();
  tabs.add(tabsArray);
}
```

### Steps to reproduce

Just execute the snippet above.

### Environment

Vaadin version(s): Vaadin 24.0.0.beta1
OS: Windows 10

### Browsers

_No response_

### PR DESCRIPTION
**Pull request by bwajtr**
## Description

A call to Tabs#add(Component...) causes ClassCastException. Even though the method is deprecated, it shouldn't throw a ClassCastException.
Component[] cannot be cast to Tabs[] -> a new array has to be created and items copied, Arrays.copyOf() seems to be the fastest of the available options.
Furthermore, in the change,  Assert.assertThat is already deprecated -> replaced that with MatcherAssert.assertThat in the test.

Fixes #4680

## Type of change

- [x] Bugfix
- [ ] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/contributing/overview
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

### CODE REVIEW

File: vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java [hunk 1]
```diff
@@ -22,11 +22,12 @@
 import org.junit.Test;
 import org.junit.rules.ExpectedException;
 
+import com.vaadin.flow.component.Component;
 import com.vaadin.flow.component.tabs.Tab;
 import com.vaadin.flow.component.tabs.Tabs;
 
 import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertThat;
+import static org.hamcrest.MatcherAssert.assertThat;
```
Comment (ugur-vaadin) [inline vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java:30]: Is this change intended?

File: vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java [hunk 2]
```diff
@@ -205,4 +206,13 @@ public void removeTabInTabsWithoutSelection() {
         Assert.assertNull("should not change selected tab",
                 tabs.getSelectedTab());
     }
+
+    @Test
+    public void addTabsAsArrayDoesNotCauseClassCastException() {
```
Comment (ugur-vaadin) [inline vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java:211]: Maybe the naming can be more specific like `addTabsAsComponentArray_noClassCastExceptionIsThrown `

File: vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java (hunk 1 above)
Comment (bwajtr) [reply vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java:30]: Yes, org.junit.Assert.assertThat is deprecated, so I replaced it with org.hamcrest.MatcherAssert.assertThat to get rid of the warnings

File: vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java (hunk 2 above)
Comment (bwajtr) [reply vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java:211]: Will do, thanks

File: vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java (hunk 1 above)
Comment (ugur-vaadin) [reply vaadin-tabs-flow-parent/vaadin-tabs-flow/src/test/java/com/vaadin/flow/component/tabs/tests/TabsTest.java:30]: Oh, alright.

**Review by ugur-vaadin — APPROVED** [review]

**Pull request merged**